src/pypromice/process/L0toL1.py

```diff
@@ -257,6 +257,41 @@ def getPressDepth(z_pt, p, pt_antifreeze, pt_z_factor, pt_z_coef, pt_z_p_coef):
     
     return z_pt_cor, z_pt
 
+
+def interpTemp(temp, var_configurations, max_interp=pd.Timedelta(12,'h')):
+    '''Clip and interpolate temperature dataset for use in corrections
+    
+    Parameters
+    ----------
+    temp : `xarray.DataArray`
+        Array of temperature data
+    vars_df : `pandas.DataFrame`
+        Dataframe to retrieve attribute hi-lo values from for temperature clipping
+    max_interp : `pandas.Timedelta`
+        Maximum time steps to interpolate across. The default is 12 hours.
+    
+    Returns
+    -------
+    temp_interp : `xarray.DataArray`
+        Array of interpolatedtemperature data
+    '''
+    # Determine if upper or lower temperature array
+    var = temp.name.lower()
+    
+    # Find range threshold and use it to clip measurements
+    cols = ["lo", "hi", "OOL"]
+    assert set(cols) <= set(var_configurations.columns)
+    variable_limits = var_configurations[cols].dropna(how="all")
+    temp = temp.where(temp >= variable_limits.loc[var,'lo'])
```

Comment:
The interpolation is basically interpolating the values just above low and above high.
It is basically a clipping.
The following code is however not limiting the period length

```python
temp = temp.clip(variable_limits.loc[var,'lo'], variable_limits.loc[var,'hi'])
```

Reply:
Hmm. Your solution didn't work for me. 

This is what I expect to produce, using the routines we currently have:
```
In [1]:  temp = temp.where(temp >= variable_limits.loc[var,'lo'])
In [2]:  temp = temp.where(temp <= variable_limits.loc[var, 'hi'])
In [3] list(temp.sel(time='2022-11-09').values)
Out[3]: 
[-49.0, nan,
 nan, nan,
 nan, nan,
 nan, nan,
 -49.95, nan,
 nan, nan,
 -48.58, -49.28,
 -48.55, -47.2,
 -46.22, -45.37,
 -44.85, -44.15,
 -43.7, -43.23,
 -43.02, -42.93]
```

This is what happens with your solution when I run it:

```
In [1]: temp = temp.clip(variable_limits.loc[var,'lo'], variable_limits.loc[var,'hi'])
In [2]: list(temp.sel(time='2022-11-09').values)
Out[2]: 
[-49.0, 40.0,
 40.0, 40.0,
 40.0, 40.0,
 40.0, 40.0,
 -49.95, 40.0,
 40.0, 40.0,
 -48.58, -49.28,
 -48.55, -47.2,
 -46.22, -45.37,
 -44.85, -44.15,
 -43.7, -43.23,
 -43.02, -42.93]
```

Instead of clipping everything outside the range to `nan`, it is re-assigning it as `40.0`. Can you either provide a more full solution or can we stick with our current clipping solution?

Reply:
At least in this situation, we should remove the OOL data, because we see that the values are completely wrong.
It is different from the clipping, for example, of RH (with regard to water), which, if measured at 105%, can be replaced by a more meaningful 100%.
In general, I am against clipping (replacing OOL values by the limit that was crossed). If it's outside of the acceptable range, then the value is not reliable and it should be replaced by NaN.

Reply:
Does anything need to be changed in the current solution then @BaptisteVandecrux?